### DISCUSSION TITLE
Will turning on mobile backup from both iPhone and iPad cause issues?

### DISCUSSION THREAD

**Discussion by charles-997** [immich-app/immich#12477] [Q&A] [answered]
I’m new to Immich and I am doing my initial upload. Can I use the backup function on both my iPad and iPhone simultaneously? I understand Immich will not upload duplicates.  Will using both devices cause any unforeseen issues?

Both devices use the same iCloud library with the same images.
My initial upload is taking a few days, so I want to leave my iPad at home to upload faster.

**Comment by alextran1502** [comment, marked as answer]
It won't cause issue since already uploaded photos will be rejected

**Reply by xiaosongz** [reply]
I noticed that current mobile clients must fully download each iCloud photo or video and then upload it to the Immich server before duplicates are rejected. Is there a more efficient way to handle the initial sync? For example, could the client compute a local hash (e.g., SHA256) for each asset—even if it’s still in iCloud—and then query the server to see if that hash already exists? If the file is already present, the client could skip both download and upload, saving potentially gigabytes of unnecessary data transfer.
